pipelines/scripts/copy_key_vault_secrets.py

```diff
@@ -0,0 +1,55 @@
+from azure.identity import AzureCliCredential
+from azure.keyvault.secrets import SecretClient
+import argparse
+from typing import Set
+import logging
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+class KeyVaultManager():
+    """
+        Wrapper class to handle the interaction with a key vault
+    """
+    def __init__(self, vault_name: str):
+        credential = AzureCliCredential()
+        self.client = SecretClient(vault_url=f"https://{vault_name}.vault.azure.net", credential=credential)
+    
+    def create_secret(self, secret_name: str, secret_value: str):
+        return self.client.set_secret(secret_name, secret_value)
+
+    def get_secret(self, secret_name: str):
+        return self.client.get_secret(secret_name)
+    
+    def get_secrets(self) -> Set[str]:
+        return {x.name for x in self.client.list_properties_of_secrets()}
+
+
+def copy_secrets_between_key_vaults(source_vault: str, target_vault: str):
```

Comment:
Don't you want to add some error handling in this function ? like a try/except to log errors if the secrets fails ?
Maybe also having a dry run to preview which secrets would be copied ?

Reply:
I don't think there would be much value. The logs print the current secret being copied, and if something goes wrong it'll crash anyway, and whoever is running the script will be able to see the secret that caused it from the logs